hudi-client/src/main/java/org/apache/hudi/client/utils/ClientUtils.java

```diff
@@ -37,6 +37,8 @@
   public static HoodieTableMetaClient createMetaClient(JavaSparkContext jsc, HoodieWriteConfig config,
       boolean loadActiveTimelineOnLoad) {
     return new HoodieTableMetaClient(jsc.hadoopConfiguration(), config.getBasePath(), loadActiveTimelineOnLoad,
-        config.getConsistencyGuardConfig(), Option.of(new TimelineLayoutVersion(config.getTimelineLayoutVersion())));
+        config.getConsistencyGuardConfig(),
+        Option.ofNullable((config.getTimelineLayoutVersion() != null)
+            ? new TimelineLayoutVersion(config.getTimelineLayoutVersion()) : null));
```

Comment:
can the default in HoodieWriteConfig be `Option.empty` instead of null? Then, we can simply make `config.getTimelineLayoutVersion()` return a Option directly

Reply:
Yeah, this can be simplified. HoodieWriteConfig's TimelineLayoutVersion will not be null as it has default. Will change based on that.